packages/dds/tree/src/feature-libraries/editable-tree/editableTreeContext.ts

```diff
@@ -83,7 +89,34 @@ export class ProxyContext implements EditableTreeContext {
         }
         cursor.free();
         this.forest.anchors.forget(destination);
+        return proxifyField(
+            rootSchema,
+            symbolFromKey(rootFieldKey),
+            targets,
+            true,
+        ) as UnwrappedEditableField;
+    }
+
+    // This code is kept independent of `get unwrappedRoot()` (i.e. duplicated) on purpose,
+    // as it will probably differ a lot in editing case.
```

Comment:
Why not just have a private method that gets the root and takes a Boolean for unwrap or not, and have both this and unwrappedRoot call it?

Part of the motivation for making roots work the way they do is editing them should not be significantly different from editing any other field (the have a field schema, and children just like other fields), so I don't think duplication here should be necessary. Once we have proper types for fields that are not unwrapped, those should work fine for editing the root.

Reply:
As more I look at it (and also already having an implementation for sequences, where it didn't change at all), as more I agree on that it should not differ so much. Ok, I'll fix this.